packages/kit/src/runtime/server/page/render.js

```diff
@@ -353,17 +353,32 @@ export async function render_response({
 	// add the content after the script/css links so the link elements are parsed first
 	head += rendered.head;
 
+	const html = options.app_template({
+		head,
+		body,
+		assets,
+		nonce: /** @type {string} */ (csp.nonce)
+	});
+
 	// TODO flush chunks as early as we can
-	const html =
+	const transformed =
 		(await resolve_opts.transformPageChunk({
-			html: options.app_template({ head, body, assets, nonce: /** @type {string} */ (csp.nonce) }),
+			html,
 			done: true
 		})) || '';
 
+	if (__SVELTEKIT_DEV__) {
+		if (transformed.split('<!--').length < html.split('<!--').length) {
+			console.warn(
+				'\u001B[1m\u001B[31mRemoving comments in transformPageChunk can have undesirable effects\u001B[39m\u001B[22m'
```

Comment:
perhaps we should say "break Svelte's hydration" rather than "have undesirable effects" as the latter is rather vauge?

Reply:
```suggestion
				'\u001B[1m\u001B[31mRemoving comments in transformPageChunk can break Svelte\'s hydration\u001B[39m\u001B[22m'
```

Reply:
yeah, i like that